File: lab spring25/python a2/taskC.py
import sys 
sys.stdin = open("input.txt","r")

def solve():
    n,k = list(map(int,input().split()))
    arr = list(map(int,input().split()))

    mxlen = 0
    sm = 0
    i,j=0,0

    while j<n:
        if sm+arr[j]<=k:
            sm+=arr[j]
            mxlen = max(mxlen,j-i+1)
            j+=1
        elif sm!=0:
            sm-=arr[i]
            i+=1
        else:
            i+=1
            j+=1
    
    print(mxlen)

t = 1
t = int(input())
while t:
    solve()
    t-=1
    print()

File: lab spring25/assignment1/taskC.cpp
#include <iostream>
using namespace std;
#define br cout << '\n';

void solve(){
	int n,k;
	cin >> n >> k;
	int arr[n];
	for(int i=0;i<n;i++) cin >> arr[i];
	for(int i=0;i<k;i++) cout << arr[k-i-1] << ' ';br;
}

int main(){
	freopen("input.txt","r",stdin);
	solve();
}
